android/src/main/java/com/swmansion/reanimated/nodes/ArrayFromNode.java

```diff
@@ -0,0 +1,41 @@
+package com.swmansion.reanimated.nodes;
+
+import com.facebook.react.bridge.JavaOnlyArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableArray;
+import com.swmansion.reanimated.NodesManager;
+import com.swmansion.reanimated.Utils;
+
+public class ArrayFromNode extends Node {
+
+    private final int[] mArrayFrom;
+
+    public ArrayFromNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
+        super(nodeID, config, nodesManager);
+        mArrayFrom = Utils.processIntArray(config.getArray("arrayFrom"));
+    }
+
+    @Override
+    protected Object evaluate() {
+        JavaOnlyArray res = new JavaOnlyArray();
+        for (int el: mArrayFrom) {
+            Object value = mNodesManager.findNodeById(el, Node.class).value();
+
+            // it's here to match objc implementation that can handle any type
+            if (value == null) {
+                res.pushNull();
+            } else if (value instanceof Number) {
+                res.pushDouble((Double) value);
+            } else if (value instanceof String) {
+                res.pushString((String) value);
+            } else if (value instanceof Boolean) {
+                res.pushBoolean((Boolean) value);
+            } else if (value.getClass().isArray()) {
```

Comment:
Are you certain that `value.getClass().isArray()` returns `true` when `value` is instance of `WritableArray`?

Reply:
Not sure, will check!

Reply:
it's not working, sooo... `(value.getClass().isArray() || value instanceof ReadableArray)`?

Reply:
maybe simply `value instanceof ReadableArray)`?